### ISSUE TITLE
Equations not wrapping in LaTeX->PDF conversion

### ISSUE THREAD

**Issue by ianhbell** [jupyter/nbconvert#275]
I work with sympy quite a bit, and one issue that I am running into is that large equations are output as tiny text all on one line, rather than being wrapped like in the HTML MathJax output. Is there any way of automatically making the LaTeX conversion do something nice like MathJax does?

Here is a nicer screenshot from the Jupyter notebook:
![capture](https://cloud.githubusercontent.com/assets/11966765/14115663/267a458a-f599-11e5-8a7b-738eafdea74b.PNG)

And the unreadable all-on-one-line output in the PDF:
![capture2](https://cloud.githubusercontent.com/assets/11966765/14115694/4bfc82a0-f599-11e5-9f1e-224675b9d247.PNG)

**Comment by ianhbell** [comment]
I have a hacked together solution based on parsing the intermediate .tex file ( better solutions welcomed) :

1. Write a script that converts equations like ``\[ ... \]`` to ``\begin{breqn}... \end{breqn} ``
2. Add ``\usepackage{breqn}`` to preamble

``` python
oldName = "old.tex"
newName = "new.tex"
with open(oldName,'r') as fp:
    contents = fp.read()
    contents = contents.replace(r'\[','\\begin{dmath}').replace(r'\]','\\end{dmath}').replace(r'\begin{document}',r"""\usepackage{breqn}\n    \begin{document}""")
    with open(newName,'w') as fpout:
        fpout.write(contents)
```

One of the parts of the problem is that pandoc doesn't use equation environment, it outputs the equations coming from sympy as `` \[ ... \] `` , and it doesn't seem like there is anything that can be done about that in jinja

**Comment by minrk** [comment]
I'm not sure what options are available to us, but we may be able to do some configuration of pandoc. Perhaps talking directly to pandoc folks about the desired tex result is the best course right now, then we can see if there's a way to get what we want on the nbconvert side.

**Comment by mgeier** [comment]
@ianhbell Can you please provide a link to an example notebook?

I don't think pandoc is involved if the equation is in a code cell output ...
Probably something can be done on the SymPy side?

**Comment by ianhbell** [comment]
How about this MWE? Demonstrates the exact problem:
```
from IPython.display import display, Math, Latex
from IPython.core.display import display_html
from sympy import *
init_session(quiet=True)
init_printing()
x = symbols('x')
arg = x
for i in range(120):
    arg += x**i
arg
```

**Comment by mgeier** [comment]
Thanks for the example.

I don't have a solution (yet), but here are a few thoughts:

It looks like SymPy generates output surrounded by double dollar signs, e.g. `$$x^2$$`, which is translated by `nbconvert` to `\[x^2\]` via `markdown2latex`. It looks like pandoc is involved after all, although I don't really know why LaTeX blocks are processed with a Markdown filter ... ?

The function `sympy.init_printing()` has a `latex_mode` argument (which should be one of `{'inline'|'plain'|'equation'|'equation*'}`), but this seemed to have no effect when I tried it in The Notebook.

Anyway, if you get SymPy to produce  `$x^2$` (with single dollar signs) or `\(x^2\)`, the line breaks should work as expected. However, some Symbols (like e.g. `\int`) will look ugly (because they are shrunk to fit the line height), therefore you should probably switch to `\displaystyle`. In the end, you should try to get SymPy to produce something like  `$\displaystyle x^2$`.

Finally,  to get rid of the hanging indent, you might want to prepend `\noindent`. I don't think this can be added in the SymPy output, because it would break the MathJax display. But it should be easy to add this in the Jinja template.

**Comment by jsbangsund** [comment]
I'm not working with sympy, but I've had issues with inline equations in markdown cells upon converting latex -> pdf. If I change the same cell to "raw" everything works fine, but as markdown I get this error, trying to add an additional $ sign, when I only used one to make the equation inline:

LaTeX Warning: No \author given.

[1{C:/ProgramData/MiKTeX/2.9/pdftex/config/pdftex.map}]
("C:\Program Files\MiKTeX 2.9\tex\latex\psnfss\t1pcr.fd")
! Missing $ inserted.
<inserted text> 
                $
l.354 ...l the Boltzmann equation: \$ S = k\_B \ln
                                                  \Omega \$
? 
! Emergency stop.
<inserted text> 
                $
l.354 ...l the Boltzmann equation: \$ S = k\_B \ln
                                                  \Omega \$
!  ==> Fatal error occurred, no output PDF file produced!

Anybody have a workaround?

**Comment by mgeier** [comment]
Do you happen to have spaces on the right of the opening and/or left of the closing dollar sign?
If yes, that may be the problem, see http://pandoc.org/MANUAL.html#math.
If not, please provide your original Markdown text.

**Comment by jsbangsund** [comment]
@mgeier That was it exactly. Thank you.

Changing $ S = k_B \ln \Omega $ to $S = k_B \ln \Omega$ solved the issue.

**Comment by mgeier** [comment]
https://github.com/sympy/sympy/pull/15625 will fix this issue.

**Comment by mgeier** [comment]
https://github.com/sympy/sympy/pull/15625 was merged and will be available in the next SymPy release.

@ianhbell Can you please check if that works for you?

**Comment by ianhbell** [comment]
How can I install the dev version of sympy?  I can check if you can point me in the right direction.

**Comment by ibell** [comment]
Works perfectly with my example above:
![image](https://user-images.githubusercontent.com/1859947/52030713-5938b280-24d6-11e9-9384-a699fc312bc4.png)

**Comment by ibell** [comment]
Sorry, @ianhbell and @ibell are both me

**Comment by t-makaro** [comment]
I can confirm that this is better, but still not perfect.

Good:
![image](https://user-images.githubusercontent.com/18018386/57884822-6dcb5d00-77de-11e9-90de-e78b09aed76d.png)

Less so:

![image](https://user-images.githubusercontent.com/18018386/57884746-3eb4eb80-77de-11e9-93d5-5e4d03af801f.png)

![image](https://user-images.githubusercontent.com/18018386/57884757-470d2680-77de-11e9-9950-4511c23a491d.png)

Bad:

![image](https://user-images.githubusercontent.com/18018386/57884768-4d9b9e00-77de-11e9-9335-1e30c7a15298.png)

**Comment by mgeier** [comment]
Yeah, I've seen problems with large sub-expressions, especially when using matrices.

@t-makaro Does this even work with "normal" LaTeX?

**Comment by MSeal** [comment]
Do you think there's anything we should change here @mgeier @t-makaro ? Or should we close this issue for now?

**Comment by mgeier** [comment]
You should consider using `\raggedright`, as in the PR I mentioned above.

Here's an example where this makes a difference: https://github.com/AudioSceneDescriptionFormat/asdf/blob/master/doc/quaternions.ipynb

It's the first cell in the section "Combining all Axes".

Or maybe you have another idea how to improve the line-breaking behavior in this case?

**Comment by meeseeksmachine** [comment]
This issue has been mentioned on **Jupyter Community Forum**. There might be relevant details there:

https://discourse.jupyter.org/t/print-preview-of-jupiter-notebook-sympy-output/5205/2
